tests/unit/test_git_hooks.py

```
@@ -0,0 +1,247 @@
+            env["PATH"] = f"{bindir}{os.pathsep}{env.get('PATH', '')}"
+        return subprocess.run(["sh", str(script)], cwd=str(cwd), env=env).returncode
+
+    def _git(self, cwd, *args) -> None:
+
+        subprocess.run(["git", *args], cwd=str(cwd), check=True, capture_output=True)
+
+    def test_guard_blocks_branch_merged_into_default(self, tmp_path):
+        """A branch fully contained in a moved-on origin/main is blocked."""
+
+        remote = tmp_path / "remote.git"
+        subprocess.run(["git", "init", "-q", "--bare", str(remote)], check=True)
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)